### PR TITLE
13472(mattermost-server) Migrate 'components/more_channels' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by ShehryarShoukat96**
#### Summary
This PR migrates 'components/more_channels' module and associated tests to TypeScript.

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/13472

### CODE REVIEW

**Comment by hanzei** [comment]
Hi @ShehryarShoukat96,

Thanks for the PR! Could you please fix the failing CI? After that, I will request developer reviews for your PR.

**Comment by ShehryarShoukat96** [comment]
@hanzei Hi, sorry for the delay. I have fixed the CI error. You can remove the work in progress tag and review this PR. Thanks

File: components/more_channels/more_channels.tsx [hunk 1]
```diff
@@ -68,15 +84,15 @@ export default class MoreChannels extends React.PureComponent {
         if (this.props.bodyOnly) {
             this.handleExit();
         }
-    }
+    };
```
Comment (devinbinnie) [inline components/more_channels/more_channels.tsx:87]: We don't require semi-colons at the end of these arrow functions, just for the future.

File: components/more_channels/more_channels.tsx [hunk 2]
```diff
@@ -85,68 +101,67 @@ export default class MoreChannels extends React.PureComponent {
...
-            } else {
-                browserHistory.push(getRelativeChannelURL(teamName, channel.name));
-                this.handleHide();
-            }
-
-            if (done) {
-                done();
-            }
-        });
+        const result = await actions.joinChannel(currentUserId, teamId, channel.id) as { error: any };
```
Comment (devinbinnie) [inline components/more_channels/more_channels.tsx:112]: ❤️   this

File: components/more_channels/more_channels.test.tsx [hunk 3]
```diff
@@ -98,16 +132,19 @@ describe('components/MoreChannels', () => {
         expect(wrapper.state('searching')).toEqual(false);
 
         // on componentDidMount
-        expect(wrapper.instance().props.actions.getChannels).toHaveBeenCalledTimes(1);
-        expect(wrapper.instance().props.actions.getChannels).toHaveBeenCalledWith(wrapper.instance().props.teamId, 0, 100);
+        const instance = wrapper.instance() as MoreChannels;
```
[the remark this answers is not in the document]
Comment (ShehryarShoukat96) [reply components/more_channels/more_channels.test.tsx:135]: @devinbinnie Thanks

**Comment by ShehryarShoukat96** [comment]
@devinbinnie Thanks for the comments. Can you review it again?

**Pull request merged**